components/core/src/clp_s/JsonParser.cpp

```diff
@@ -353,10 +349,7 @@ void JsonParser::parse_line(ondemand::value line, int32_t parent_node_id, std::s
                 break;
             }
             case ondemand::json_type::string: {
-                auto raw_json_token = line.raw_json_token();
-                std::string value
-                        = std::string(raw_json_token.substr(1, raw_json_token.rfind('"') - 1));
-
+                std::string_view value = line.get_string(true);
```

Comment:
Right, the behaviour with true specifically is that it will replace any invalid unicode codepoint with a placeholder unicode character. This ensures that this method always returns valid utf-8, but can sometimes modify the original input.

The way I see this is that if a user wants to ingest their data despite it containing some invalid unicode this is really one of the only good ways of doing it automatically.

That said, its possible that some users would prefer failing ingestion so that they get direct feedback that they're producing invalid JSON. I could add a flag for this in this PR, or create a github issue to track this feature to add it later if you want?

Also, for what its worth, I talked to @kirkrodrigues when deciding what to do here and he agreed that the automatic utf-8 fix is probably desirable for large deployments.